src/sentry/snuba/sessions_v2.py

```diff
@@ -324,15 +324,49 @@ def __init__(
         else:
             column_resolver = resolve_column
 
-        conditions = [resolve_condition(c, column_resolver) for c in snuba_filter.conditions]
+        condition_cols_set = set(CONDITION_COLUMNS)
+        self.conditions = self._get_conditions(
+            snuba_filter.conditions, column_resolver, condition_cols_set
+        )
+
         filter_keys = {
             resolve_filter_key(key): value for key, value in snuba_filter.filter_keys.items()
         }
 
         self.aggregations = snuba_filter.aggregations
-        self.conditions = conditions
         self.filter_keys = filter_keys
 
+    def _get_conditions(self, sn_filter_conditions, col_resolver, con_cols, max_nested_depth=5):
+        # Conditions in nested lists with depth >max_nested_depth are ignored.
+        def helper(cons, depth):
+            if depth <= 0:
+                return []
```

Comment:
This piece here silently drops conditions, to not return error responses. We may want to consider raising an error, or logging something with Sentry.